### ISSUE TITLE
🤖 Baseline ML Models

### ISSUE THREAD

**Issue by CSFelix** [CSFelix/Sales-Prediction#10]
Usefull links

- [Intro to Keras with breast cancer data[ANN]](https://www.kaggle.com/code/thebrownviking20/intro-to-keras-with-breast-cancer-data-ann/notebook)

- [Intro to Recurrent Neural Networks LSTM | GRU](https://www.kaggle.com/code/thebrownviking20/intro-to-recurrent-neural-networks-lstm-gru/notebook)

**Comment by CSFelix** [comment]
Now, even though RNNs are quite powerful, they suffer from Vanishing Gradient Problem which hinders them from using long term information, like they are good for storing memory 3-4 instances of past iterations but larger number of instances don't provide good results so we don't just use regular RNNs. Instead, we use a better variation of RNNs: Long Short Term Networks (LSTM).

**Comment by CSFelix** [comment]
Components of LSTMs

- Forget Gate “f” ( a neural network with sigmoid);
- Candidate layer “C"(a NN with Tanh);
- Input Gate “I” ( a NN with sigmoid );
- Output Gate “O”( a NN with sigmoid);
- Hidden state “H” ( a vector );
- Memory state “C” ( a vector);

- Inputs to the LSTM cell at any step are Xt (current input) , Ht-1 (previous hidden state ) and Ct-1 (previous memory state);
- Outputs from the LSTM cell are Ht (current hidden state ) and Ct (current memory state);

**Comment by CSFelix** [comment]
There's no need to use **LSTM and GRU** since the dataset is not a Time Series at all.

**Comment by CSFelix** [comment]
Models

* [x] Linear Regression
* [x] Random Forest Regressor
* [x] XGBoost
